README.md

```diff
@@ -1,15 +1,121 @@
-# Hyperlane Warp Route
+# Hyperlane Warp Routes
 
-This repo contains the base Hyperlane ERC20 and ERC721 tokens (HypERC20 and HypERC721). These tokens extend the base standards with an additional `transferRemote` function. Warp Routes are way of arranging these contracts to make existing assets interchain. Read more about Warp Routes and how to deploy your own at [Warp API docs](https://docs.hyperlane.xyz/docs/developers/warp-api).
+This repo contains contracts, deployment, and SDK tooling for Hyperlane Warp Routes. 
 
-## Versions
+## Warp Route Architecture
 
-| Git Ref | Release Date | Notes |
-| ------- | ------------ | ----- |
-| [audit-v2-remediation]() | 2023-02-15 | Hyperlane V2 Audit remediation |
-| [main]() | ~ | Bleeding edge |
+A *warp route* is a collection of [`TokenRouter`](./contracts/libs/TokenRouter.sol) contracts deployed across a set of Hyperlane chains. These contracts leverage the `Router` pattern to implement access control and routing logic for remote token transfers. These contracts send and receive [`Message`](./contracts/libs/Message.sol)s which encode payloads containing a transfer `amount` and `recipient` address.
+
+```mermaid
+%%{ init: {
+  "theme": "neutral",
+  "themeVariables": {
+    "mainBkg": "#025AA1",
+    "textColor": "white",
+    "clusterBkg": "beige"
+  },
+  "themeCSS": ".edgeLabel { color: black }"
+}}%%
+
+graph LR
+    subgraph "Ethereum"
+        HYP_E[TokenRouter]
```

Comment:
IDK if you were planning on doing this in a follow on change, but should this be tokenRouter, or the specific instances (HypNative/HypErc20Collateral)